napari_btrack/track.py

```diff
@@ -1,24 +1,144 @@
+                        0, 0, 0, 0, 0, 1],
+            H=[1, 0, 0, 0, 0, 0,
+               0, 1, 0, 0, 0, 0,
+               0, 0, 1, 0, 0, 0],
+            P=[0.1, 0, 0, 0, 0, 0,
+                0, 0.1, 0, 0, 0, 0,
+                0, 0, 0.1, 0, 0, 0,
+                0, 0, 0, 1, 0, 0,
+                0, 0, 0, 0, 1, 0,
+                0, 0, 0, 0, 0, 1],
+            G=[0.5, 0.5, 0.5, 1, 1, 1],
+            R=[1, 0, 0,
+                0, 1, 0,
+                0, 0, 1],
+            # fmt: on
+        )
+    )
+
+    @classmethod
+    def get_scaled_matrix(
+        cls, name: str, sigma: float, use_cell_config: bool = True
+    ) -> List[float]:
+        """Returns the scaled version (i.e. the unscaled matrix multiplied by sigma)
+        of the matrix.
+
+        Keyword arguments:
+        name -- the matrix name (can be one of A, H, P, G, R)
+        sigma -- the factor to scale the matrix entries with
+        cell -- whether to use cell config matrices or not (default true)
+        """
+        if name == "A":
+            if use_cell_config:
+                name = "A_cell"
+            else:
+                name = "A_particle"
+        return (np.asarray(cls().unscaled_matrices[name]) * sigma).tolist()
+
+    @classmethod
+    def get_sigma(cls, name: str, scaled_matrix: npt.NDArray[np.float64]) -> float:
+        """Returns the factor sigma which is the multiplier between the given scaled
+        matrix and the unscaled matrix of the given name.
+
+        Note: Calculation relies on the top-left entry of the matrix.
+
+        Keyword arguments:
+        name -- the matrix name (can be one of A, H, P, G, R)
+        scaled_matrix -- the scaled matrix to find sigma from.
+        """
+        if name == "A":
+            name = "A_cell"  # doesn't matter which A we use here, as [0][0] is the same
+        return scaled_matrix[0][0] / cls().unscaled_matrices[name][0]
+
+
+def run_tracker(
+    segmentation: Union[napari.layers.Image, napari.layers.Labels],
+    tracker_config: TrackerConfig,
+) -> Tuple[npt.NDArray, dict, dict]:
+    """
+    Runs BayesianTracker with given segmentation and configuration.
+    """
     with btrack.BayesianTracker() as tracker:
-        # configure the tracker using a config file
-        tracker.configure_from_file(config_file_path)
+        tracker.configure(tracker_config)
         tracker.max_search_radius = 50
 
         # append the objects to be tracked
-        tracker.append(objects)
+        segmented_objects = segmentation_to_objects(segmentation.data)
+        tracker.append(segmented_objects)
 
         # set the volume
-        tracker.volume = ((0, 1600), (0, 1200), (-1e5, 64.0))
+        segmentation_size = segmentation.level_shapes[0]
+        # btrack order of dimensions is opposite of napari
+        tracker.volume = (
+            (0, segmentation_size[2]),
+            (0, segmentation_size[1]),
+            (-1e5, segmentation_size[0]),
```

Comment:
I don't think this is correct.

Napari (and others) store the image data as T(Z)XY and the btrack volume is set as the limits on XY(Z).  As you have it, it's using the T dimension as a Z dimension.

The logic needs to be something like:

```python
volume = tuple([(0, s) for s in segmentation_size[1:][::-1]])
```

to deal with the fact that some data are 2D and some are 3D.

Reply:
In reality, we may need to consider the napari world coordinates and such at a later date -- but this should suffice for now.

Reply:
Opened #32 so we dont' forget to consider this in the future.